### PR TITLE
Update matchms to 0.17.0

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`matchms`](https://bioconda.github.io/recipes/matchms/README.html): **0.16.0** &rarr; **0.17.0**

[install with bioconda](http://bioconda.github.io/recipes/matchms/README.html) [Conda](https://anaconda.org/bioconda/matchms/files)

Info | Link or Description
-----|--------------------
Recipe | [`recipes/matchms`](https://github.com//bioconda/bioconda-recipes/tree/bump/matchms/recipes/matchms) (click to view/edit other files)
Summary | Python library for fuzzy comparison of mass spectrum data and other Python objects
Home | [https://github.com/matchms/matchms](https://github.com/matchms/matchms)
Releases |[https://pypi.org/pypi/matchms/json](https://pypi.org/pypi/matchms/json)
Recipe Maintainer(s) | @RECETOX

***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Review by rpetit3 — APPROVED** [review]

**Comment by rpetit3** [comment]
@RECETOX I did a quick skim of the new release and didn't look like dependencies changed. Are there any changes needed?

**Comment by maximskorik** [comment]
@rpetit3 the new version includes new features based on built-in modules, so there aren't any new dependencies.

On another note, your message reminded me that using an organization tag in `recipe-maintainers` doesn't work as we haven't received any notification about your comment. Could you please merge https://github.com/bioconda/bioconda-recipes/pull/37434 into this branch before merging?

**Comment by maximskorik** [comment]
@BiocondaBot please merge

**Pull request merged**
